### DISCUSSION TITLE
Why is the entire address for a mailto link &-encoded?

### DISCUSSION THREAD

**Discussion by Charles-Gagnon** [markedjs/marked#2142] [Q&A]
[demo](https://marked.js.org/demo/?outputType=lexer&text=test%40gmail.com&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

An e-mail address such as `test@gmail.com` is converted into a mailto link such as `mailto:&#116;&#101;&#x73;&#x74;&#64;&#x67;&#x6d;&#x61;&#105;&#x6c;&#x2e;&#x63;&#111;&#x6d;`

Is there any documentation about why every character is encoded like this?

**Comment by UziTech** [comment]
The `mangle` [option](https://marked.js.org/using_advanced) is `true` by default changing it to `false` will prevent encoding email addresses.

**Reply by Charles-Gagnon** [reply]
Gotcha, thank you!

Do you have any context on why this would be enabled by default? Is there some potential issue that setting it to false would expose?

**Reply by UziTech** [reply]
I think it is to prevent bots from crawling a website and spamming the email address. It probably isn't as useful today as when marked was first created since bots are probably smart enough to get the email address anyway. More or less a product of backwards compatibility.
